### PR TITLE
chore(libmc.mk): get architecture from script

### LINKED ISSUE #1616: find a way to support m1 & intel macs as build target
#### Why is this needed:
We do not have a way to identify which architecture to use when we get a match for `x86_64-apple-darwin18` since both the intel & m1 macs are a match. We need to find a way to identify m1 macs so we can set the target to `aarch64-apple-darwin`.

### PR DESCRIPTION
**Pull request by DieHard073055**
#### What kind of PR is this?:

/kind chore

#### What this PR does / why we need it:
Figure out if the host architecture is apple m1 mac or intel.
#### Which issue(s) does this PR fixes?:

Fixes #1616

### CODE REVIEW

File: depends/libmc_target.sh [hunk 1]
```diff
@@ -0,0 +1,31 @@
+#! /bin/bash
```
Comment (wafflespeanut) [inline depends/libmc_target.sh:1]: If this works as expected when building for multiple targets, then we could rename it to `rust_targets.sh` and use it for `libain.mk` as well.

Comment (DieHard073055) [reply depends/libmc_target.sh]: updated filename.

Comment (wafflespeanut) [reply depends/libmc_target.sh]: And using it in `libain.mk`?

Comment (DieHard073055) [reply depends/libmc_target.sh]: and `libain.mk`

**Pull request merged**